### ISSUE TITLE
[Bug]: Equivalent query sometimes fails to query results on EKS with 2 cn

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#8144]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):nightly-4b2bcc2
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

when run tpcc  test on eks, for new order txn ,sometimes the txn can not query item data from item table.

<img width="408" alt="image" src="https://user-images.githubusercontent.com/47846308/221122401-7fcff61c-25aa-453a-9043-ce1f2b3e5a35.png">

but the record really exists in the table:

<img width="869" alt="image" src="https://user-images.githubusercontent.com/47846308/221122469-42291d94-0c2e-47b4-b81e-6729281c351b.png">

**Comment by ouyuanning** [comment]
i rep on my local env by load tpcc_10 data.
that seams problem is:
1、load data to block without sort by pk
2、when read from block.  if table has pk, we will use binary search to filter the batch, then we get the wrong result

i will ask some help from @jianwan0214 to fix it.

**Comment by ouyuanning** [comment]
confirm with @JackTan25 
this problem was caused by this PR https://github.com/matrixorigin/matrixone/pull/8119
@JackTan25 will help me to fix it.

**Comment by aressu1985** [comment]
fixed:nightly-1f68e29
